packages/flutter/lib/src/widgets/form.dart

```diff
@@ -76,10 +76,21 @@ class Form extends StatefulWidget {
   const Form({
     Key key,
     @required this.child,
+    @Deprecated(
+        'Use autoValidateMode parameter which provide more specific '
+        'behaviour related to auto validation. '
+        'This feature was deprecated after v1.17.1'
+    )
+    // ignore: deprecated_member_use_from_same_package
     this.autovalidate = false,
     this.onWillPop,
     this.onChanged,
+    AutoValidateMode autoValidateMode = AutoValidateMode.disabled,
   }) : assert(child != null),
+       autoValidateMode = autoValidateMode != AutoValidateMode.disabled
+           // ignore: deprecated_member_use_from_same_package
+           ? autoValidateMode : autovalidate
```

Comment:
I will probably simply this so that the autovalidate == true takes the highest priority and use autoValidateMode if autovalidate == false
```dart
autoValidateMode = autovalidate ? AutoValidateMode.always : autoValidateMode;
```